src/com/goide/refactor/GoIntroduceVariableBase.java

```diff
@@ -0,0 +1,252 @@
+    final String name = ContainerUtil.getFirstItem(suggestedNames);
+    assert name != null;
+    final List<PsiElement> newOccurrences = ContainerUtil.newArrayList();
+    PsiElement statement = new WriteCommandAction<PsiElement>(project, anchor.getContainingFile()) {
+      @Override
+      protected void run(@NotNull Result<PsiElement> result) throws Throwable {
+        PsiElement parent = anchor.getParent();
+        GoStatement declaration = GoElementFactory.createVarDeclarationStatement(project, name, expression);
+        PsiElement newLine = GoElementFactory.createNewLine(project);
+        PsiElement added = parent.addBefore(declaration, parent.addBefore(newLine, anchor));
+        if (replaceAll) {
+          for (PsiElement occurrence : occurrences) {
+            PsiElement occurrenceParent = occurrence.getParent();
+            if (occurrenceParent instanceof GoParenthesesExpr) occurrence = occurrenceParent;
+            PsiElement replaced = occurrence.replace(GoElementFactory.createVarReference(project, name));
+            if (replaced != null) newOccurrences.add(replaced);
+          }
+        }
+        else {
+          PsiElement newExpression = GoElementFactory.createVarReference(project, name);
+          PsiElement expressionParent = expression.getParent();
+          PsiElement toReplace = expressionParent instanceof GoParenthesesExpr ? expressionParent : expression;
+          toReplace.replace(newExpression);
+        }
+        result.setResult(added);
+      }
+    }.execute().getResultObject();
+
+  GoVarDefinition var = PsiTreeUtil.findChildOfType(statement, GoVarDefinition.class);
+    assert var != null;
+    editor.getCaretModel().moveToOffset(var.getIdentifier().getTextOffset());
+    GoInplaceVariableIntroducer introducer = new GoInplaceVariableIntroducer(var, editor, newOccurrences);
+    introducer.performInplaceRefactoring(suggestedNames);
+  }
+
+  @NotNull
+  private static LinkedHashSet<String> getSuggestedNames(GoExpression expression) {
+    if (expression instanceof GoCallExpr) {
+      List<String> names = ContainerUtil.newArrayList();
+      GoReferenceExpression callReference = PsiTreeUtil.getChildOfType(expression, GoReferenceExpression.class);
+      if (callReference != null) {
+        String name = StringUtil.decapitalize(callReference.getIdentifier().getText());
+        if (name.startsWith("get")) {
+          name = name.substring(3);
+        }
+        else if (name.startsWith("is")) {
+          name = name.substring(2);
+        }
+        for (int i = 0; i < name.length(); i++) {
+          if (i == 0 || (Character.isLowerCase(name.charAt(i - 1)) && Character.isUpperCase(name.charAt(i)))) {
+            names.add(StringUtil.decapitalize(name.substring(i)));
+          }
+        }
+      }
+      return ContainerUtil.newLinkedHashSet(ContainerUtil.reverse(names));
+    }
+    return ContainerUtil.newLinkedHashSet("i");
```

Comment:
ContainerUtil.singleton returns Set, but we need LinkedHashSet for InplaceVariableIntroducer.